### PR TITLE
Add duckdb for blocking during local enrichment and dedupe

### PR DESCRIPTION
**Pull request by jbothma**
If the enricher is a BulkEnricher, feed all the subject entities first, then iterate over candidate sets and see if they're matches.

### CODE REVIEW

File: zavod/zavod/runner/local_enricher.py [hunk 1]
```diff
@@ -146,17 +153,83 @@ def _traverse_nested(
...
+        enricher = LocalEnricher(context.dataset, context.cache, config)
+
+        context.log.info("Loading entities for matching...")
+        for entity in subject_view.entities():
+            enricher.load_wrapped(entity)
+
+        context.log.info("Matching candidates...")
+        for entity_idx, (entity_id, candidate_set) in enumerate(enricher.candidates()):
+            if entity_idx > 0 and entity_idx % 1000 == 0:
+                context.cache.flush()
```
Comment (jbothma) [inline zavod/zavod/runner/local_enricher.py:219]: this is probably irrelevant here

File: zavod/setup.py [hunk 2]
```diff
@@ -22,7 +22,7 @@
     zip_safe=False,
     install_requires=[
         "followthemoney == 3.7.*",
-        "nomenklatura[leveldb,tantivy] == 3.13.2",
+        "nomenklatura[leveldb] @ git+https://github.com/opensanctions/nomenklatura.git@just-index",
```
Comment (jbothma) [inline zavod/setup.py:25]: update when nomenklatura is merged

**Review by pudo — COMMENTED** [review]
I'd like to find a way to make `zavod` have a better module structure. What if we make a module like `zavod.dedupe` or `zavod.integrate` and put the contents of `dedupe.py` and the `duckdb_index.py` in there?

**Comment by pudo** [comment]
This is kinda dumb but I love genai SQL skills: https://chatgpt.com/share/6746cc41-0dbc-8006-a025-034002b83850

File: zavod/zavod/shed/duckdb_index.py [hunk 3]
```diff
@@ -0,0 +1,240 @@
...
+            WHERE token_freq is NULL
+            GROUP BY entries.field, entries.id, entries.token
+        """
+        self.con.execute(mentions_query)
+
+    def _build_stopwords(self) -> None:
+        token_freq_query = """
+            SELECT field, token, count(*) as token_freq
+            FROM entries
+            GROUP BY field, token
```
[the remark this answers is not in the document]
Comment (jbothma) [reply zavod/zavod/shed/duckdb_index.py:147]: when generating pairs, country makes the combinations explode a bit. But perhaps a staged approach could be used, e.g. generating pairs based on other tokens, and then only updating scores using fields or tokens with super high cardinality

**Pull request merged**